#295 - photoreel & updates

AleksandraKozubal opened the pull request:
Changed:
- references - removed links, added placeholders and possibility to set sex for proper placeholder choice
- about page: updated numbers, "Our mission" padding "Blumilk after hours" clickable photoreel
- case studies schema's bigger display, updatet gtb color palette and changed sitemap svgs
- news and single news page: person tags are clickable and work as basic tag, added displaying separate link set in admin page
- contact page, footer: changed color fot bigger contrast
-icons: replacet a few icons from .svg to html way of usage - left only those that we change accent color + from main page case study


It should close #295.

EwelinaSkrzypacz reviewed (commented):
1. When I run `php artisan migrate` I see error

![image](https://github.com/user-attachments/assets/1f44f7e1-71a4-4d4f-bdf6-2066aa407c3b)


After running `php artisan migrate:fresh` it works

2. After running`php artisan migrate:fresh --seed` images in reference section don't display

![image](https://github.com/user-attachments/assets/05103209-c9b4-4d59-bb9a-4486f41373f9)

When I add a reference in admin panel without photo (to see placeholder image) it works

3. Admin panel -> references -> create -> when I don't choose sex and click "Utwórz" I see error in console and reference is not created

![image](https://github.com/user-attachments/assets/0b27ed25-8b98-455c-8ffe-cd8704f5042d)

4. When I go to "O firmie" page I see error in console

![image](https://github.com/user-attachments/assets/b69979c1-191d-430f-9b30-4c4c18b972fe)


5. "O firmie" page - text under title "Nasze wartości" is smaller that text under title "Nasza misja". We want like that?

![image](https://github.com/user-attachments/assets/60b74c4a-2a2d-4671-8f02-4697447596a9)

AleksandraKozubal commented:
> 1. When I run `php artisan migrate` I see error
> 
> 
> [image](https://private-user-images.githubusercontent.com/56546832/363049272-1f44f7e1-71a4-4d4f-bdf6-2066aa407c3b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQ5OTc1NDcsIm5iZiI6MTcyNDk5NzI0NywicGF0aCI6Ii81NjU0NjgzMi8zNjMwNDkyNzItMWY0NGY3ZTEtNzFhNC00ZDRmLWJkZjYtMjA2NmFhNDA3YzNiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODMwVDA1NTQwN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQ5YjIxMDAzNTAzZWY1NDY4YTQ3MzcyZTQxMzM4ODMxMmYxM2Y0NDM5ZmM3YzgzYTY4OGQzZGIzODAwMzQwOWYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Fuqz3oHNIirQGVjVNfgZSEr2rSHzKyJEHGMGbY8565c)
> 
> After running `php artisan migrate:fresh` it works
> 
> 2. After running`php artisan migrate:fresh --seed` images in reference section don't display
> 
> 
> [image](https://private-user-images.githubusercontent.com/56546832/363049766-05103209-c9b4-4d59-bb9a-4486f41373f9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQ5OTc1NDcsIm5iZiI6MTcyNDk5NzI0NywicGF0aCI6Ii81NjU0NjgzMi8zNjMwNDk3NjYtMDUxMDMyMDktYzliNC00ZDU5LWJiOWEtNDQ4NmY0MTM3M2Y5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODMwVDA1NTQwN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTVmZmFlYTQ5YTgyNWI4ODg5N2I0NDA4YTAzNDdiMmRkNWY5MmE5YzMwY2RjMzVhZDNjMGI3MjQ4NTJjNmEzNDQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.NfxtPJBbdPMuwVLIGKiCgY0cbStS28418mhM3FSE71Q)
> 
> When I add a reference in admin panel without photo (to see placeholder image) it works
> 
> 3. Admin panel -> references -> create -> when I don't choose sex and click "Utwórz" I see error in console and reference is not created
> 
> 
> [image](https://private-user-images.githubusercontent.com/56546832/363051137-0b27ed25-8b98-455c-8ffe-cd8704f5042d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQ5OTc1NDcsIm5iZiI6MTcyNDk5NzI0NywicGF0aCI6Ii81NjU0NjgzMi8zNjMwNTExMzctMGIyN2VkMjUtOGI5OC00NTVjLThmZmUtY2Q4NzA0ZjUwNDJkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODMwVDA1NTQwN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWExOGVmOGFiOTg5Y2I4Y2VjMGEyMThhMTI1NTgyZGNhMjRhNjk5YTI3MzVlZDc5M2EwNDc5NThmNDIxY2JhMmUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.ieLmj9Xl2Etyagr589hJszq4zAF7ubi-vj4bzWusUUo)
> 
> 4. When I go to "O firmie" page I see error in console
> 
> 
> [image](https://private-user-images.githubusercontent.com/56546832/363051546-b69979c1-191d-430f-9b30-4c4c18b972fe.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQ5OTc1NDcsIm5iZiI6MTcyNDk5NzI0NywicGF0aCI6Ii81NjU0NjgzMi8zNjMwNTE1NDYtYjY5OTc5YzEtMTkxZC00MzBmLTliMzAtNGM0YzE4Yjk3MmZlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODMwVDA1NTQwN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFlZGZhMWZiOTk5MWQ1MjM1Zjc4MDIyZjliNzE4YmY0NDc2MzdjMzk0NjFkNWRlNDRiNDM5ZDE1M2NmNGY3NjkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.YqNbcPyWyWssQwKfnj4ES6sohdaGtlZtzbR6qVKoNT4)
> 
> 5. "O firmie" page - text under title "Nasze wartości" is smaller that text under title "Nasza misja". We want like that?
> 
> 
> [image](https://private-user-images.githubusercontent.com/56546832/363051790-60b74c4a-2a2d-4671-8f02-4697447596a9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQ5OTc1NDcsIm5iZiI6MTcyNDk5NzI0NywicGF0aCI6Ii81NjU0NjgzMi8zNjMwNTE3OTAtNjBiNzRjNGEtMmEyZC00NjcxLThmMDItNDY5NzQ0NzU5NmE5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODMwVDA1NTQwN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTk1MmIzMGJlYzRlOWI0MTEzMDIyM2IxY2E3NTk2Y2YyNDMxNzg1NDVkMzE1NGYxNmRlYjQ1MjU1ZGJlOTk0YmEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.uCNXz4aQOi1g4t4hmkM-p5hE7CZCKkX0TEcyXnAh9PQ)


1.That happened bc you had old database records that didn't contain 'sex' column. Since this last` fresh --seed` was done, you won't have that problem anymore.
2. 3. 4. fixed
5. No, changed that

kamilpiech97 commented on app/Filament/Resources/ReferenceResource.php:
```diff
@@ -34,14 +34,17 @@ public static function form(Form $form): Form
+                            ->label("Płeć")
+                            ->required()
+                            ->options([
+                                "male" => "Mężczyzna",
+                                "female" => "Kobieta",
```
```suggestion
                                "male" => "Mężczyzna",
                                "female" => "Kobieta",
```
What about other options like "nie chcę podawać" or smth like that.

AleksandraKozubal replied:
Do we really need that? It's just an option to make it easier to choose a placeholder image - we have male and female placeholders.

kamilpiech97 commented on resources/views/components/reference.blade.php:
```diff
@@ -1,16 +1,22 @@
+        <p>{!! "“" . $reference->description . "”" !!}</p>
     </blockquote>
     <figcaption class="mt-6 flex items-center gap-x-4">
+        @php
+            $photoPath = (file_exists(storage_path('app/public/' . $reference->photo)) && !!$reference->photo)
+                         ? asset('storage/' . $reference->photo)
+                         : null;
+            $placeholder = $reference->sex === 'female'
+                           ? asset('graphics/placeholders/female.svg')
+                           : asset('graphics/placeholders/male.svg');
+        @endphp
```
```suggestion
        @php
            $photoPath = (file_exists(storage_path('app/public/' . $reference->photo)) && !!$reference->photo)
                         ? asset('storage/' . $reference->photo)
                         : null;
            $placeholder = $reference->sex === 'female'
                           ? asset('graphics/placeholders/female.svg')
                           : asset('graphics/placeholders/male.svg');
        @endphp
```
Let's do that on backend

kamilpiech97 commented on resources/views/components/references.blade.php:
```diff
@@ -109,37 +109,49 @@
         @endisset
         @isset($main2)
+                @php
+                    $main2PhotoPath = (file_exists(storage_path('app/public/' . $references[$main2]->photo)) && !!$references[$main2]->photo)
+                                 ? asset('storage/' . $references[$main2]->photo)
+                                 : null;
+                    $main2Placeholder = $references[$main2]->sex === 'female'
+                                   ? asset('graphics/placeholders/female.svg')
+                                   : asset('graphics/placeholders/male.svg');
+                @endphp
```
Same here
